Sources/TuistGenerator/Graph/GraphLoaderCache.swift

```diff
@@ -2,7 +2,7 @@ import Basic
 import Foundation
 
 protocol GraphLoaderCaching: AnyObject {
-    var projects: [AbsolutePath: Project] { get }
+    var projects: [AbsolutePath: Project] { get set }
```

Comment:
I had to introduce this as a set so that I can mutate and remove the carthage projects if the option for carthage projects is disabled